prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,432 @@
+        try {
+            List<Entity> entities = generateEntities(extensionName, request);
+            for (Entity entity : entities) {
+                submitInternal(entity, doAsUser);
+            }
+            for (Entity entity : entities) {
+                scheduleInternal(entity.getEntityType().name(), entity.getName(), null, null);
+            }
+        } catch (FalconException | IOException e) {
+            LOG.error("Error when submitting extension job: ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
+        }
+        return new APIResult(APIResult.Status.SUCCEEDED, "Extension job submitted and scheduled successfully");
+    }
+
+    @POST
+    @Path("update/{extension-name}")
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
+    public APIResult update(
+            @PathParam("extension-name") String extensionName,
+            @Context HttpServletRequest request,
+            @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+        try {
+            List<Entity> entities = generateEntities(extensionName, request);
+            for (Entity entity : entities) {
+                super.update(entity, entity.getEntityType().name(), entity.getName(), null);
+            }
+        } catch (FalconException | IOException e) {
+            LOG.error("Error when updating extension job: ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
+        }
+        return new APIResult(APIResult.Status.SUCCEEDED, "Updated successfully");
+    }
+
+    @POST
+    @Path("validate/{extension-name}")
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
+    public APIResult validate(
+            @PathParam("extension-name") String extensionName,
+            @Context HttpServletRequest request,
+            @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+        try {
+            List<Entity> entities = generateEntities(extensionName, request);
+            for (Entity entity : entities) {
+                super.validate(entity);
+            }
+        } catch (FalconException | IOException e) {
+            LOG.error("Error when validating extension job: ", e);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
+        }
+        return new APIResult(APIResult.Status.SUCCEEDED, "Validated successfully");
+    }
+
+    private List<Entity> generateEntities(String extensionName, HttpServletRequest request)
+        throws FalconException, IOException {
+        // get entities for extension job
+        Properties properties = new Properties();
+        properties.load(request.getInputStream());
+        LOG.info("properties: " + properties.toString());
```

Comment:
Good catch. These are for prototyping debug and should be removed. I've removed all debugging logs.